tools/check_reordering.py

```diff
@@ -0,0 +1,167 @@
+            eprint(f"Symbol reordering in {mapPathToSource(file.filepath)}")
+        eprint("")
+
+        if fun_allowed:
+            eprint(colorama.Fore.LIGHTWHITE_EX +
+            "  Symbols are REORDERED!!\n"
+            "  Oh! MY GOD!!" 
+            + colorama.Style.RESET_ALL)
+            eprint("")
+
+    if len(comparisonInfo.missingFiles) != 0:
+        print(colorama.Fore.YELLOW + "  MISSING" + colorama.Style.RESET_ALL)
+
+        for file in comparisonInfo.missingFiles:
+            eprint(f"Symbols missing from {mapPathToSource(file.filepath)}")
+        eprint("")
+
+        if fun_allowed:
+            eprint(colorama.Fore.LIGHTWHITE_EX + "  Error, should (not) be in here " + colorama.Style.RESET_ALL)
+            eprint("")
+
+        eprint("Some files appear to be missing symbols. Have they been renamed or declared as static? You may need to remake 'expected'")
+
+
+def main():
+    description = "Check that globally visible symbols has not been reordered."
+    epilog = """\
+    N.B. Since this script reads the map files, it can only see globally visible symbols; in-function static symbols must be examined with other tools.
+    """
+
+    parser = argparse.ArgumentParser(description=description, epilog=epilog, formatter_class=argparse.RawTextHelpFormatter)
+    parser.add_argument("-a", "--print-all", help="Print all symbols of the section, not just non-matching.", action="store_true")
+    parser.add_argument("-n", "--no-fun-allowed", help="Remove amusing messages.", action="store_true")
+    parser.add_argument("-r", "--no-reverse-check", help="Disable looking for symbols on the expected map that are missing on the built map file.", action="store_true")
+    parser.add_argument("-s", "--section", help="Specify which section should be checked for reordered symbols. Use `all` to check all sections. Defaults to .bss", default=".bss", choices=[".text", ".data", ".rodata", ".bss", "all"])
+    args = parser.parse_args()
+
+    mapfilePath = Path("build/mm.map")
+    mapfileExpectedPath = "expected" / mapfilePath
+    reverseCheck: bool = not args.no_reverse_check
+    section: str = args.section
+
+    comparisonInfo = compareMapFiles(mapfilePath, mapfileExpectedPath, section, reverseCheck)
+    printSymbolComparison(comparisonInfo, args.print_all)
+
+    if len(comparisonInfo.badFiles) + len(comparisonInfo.missingFiles) != 0:
+        printFileComparison(comparisonInfo, not args.no_fun_allowed)
+        return 1
+
+    eprint("")
+    eprint(colorama.Fore.GREEN + "  GOOD" + colorama.Style.RESET_ALL)
+
+    if args.no_fun_allowed:
+        return 0
+
+    eprint("\n" + colorama.Fore.LIGHTWHITE_EX +
+    colorama.Back.RED + f"                                  " + colorama.Back.RESET + "\n" +
+    colorama.Back.RED + f"         CONGRATURATIONS!         " + colorama.Back.RESET + "\n" +
+    colorama.Back.RED + "{:^34}".format(f"All Global {section} is correct.") + colorama.Back.RESET + "\n" +
```

Comment:
```suggestion
    colorama.Back.RED + "{:^34}".format(f"All Global {section if section != "all" else "Symbols"} is correct.") + colorama.Back.RESET + "\n" +
```
Very minor (tbh I don't expect all to be used all that often).